### PR TITLE
#525 Make all Envelope classes abstract

### PR DESCRIPTION
**Pull request by fabriciofx**
As per #525 .

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this pull request

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#528` is now in scope.

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#528` assigned to [@llorllale](http://www.0crat.com/u/llorllale}). The budget is [fixed](http://datum.zerocracy.com/pages/policy.html#4) and it is 30 minutes. Please, read [§4](http://datum.zerocracy.com/pages/policy.html#4), [§8](http://datum.zerocracy.com/pages/policy.html#8), and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

File: src/main/java/org/cactoos/collection/CollectionEnvelope.java [hunk 1]
```diff
@@ -37,9 +37,15 @@
  * @version $Id$
  * @param <X> Element type
  * @since 0.23
+ * @checkstyle AbstractClassNameCheck (500 lines)
  */
-@SuppressWarnings("PMD.TooManyMethods")
-public class CollectionEnvelope<X> implements Collection<X> {
+@SuppressWarnings(
```
Comment (llorllale) [inline src/main/java/org/cactoos/collection/CollectionEnvelope.java]: Don't suppress this warning. There are no qulice rules against abstract classes.

File: src/main/java/org/cactoos/collection/CollectionEnvelope.java [hunk 2]
```diff
@@ -37,9 +37,15 @@
...
+    {
+        "PMD.TooManyMethods",
+        "PMD.AbstractNaming"
+    }
+)
+public abstract class CollectionEnvelope<X> implements Collection<X> {
```
Comment (llorllale) [inline src/main/java/org/cactoos/collection/CollectionEnvelope.java]: Rename to **AbstractCollectionEnvelope**

Comment (fabriciofx) [reply src/main/java/org/cactoos/collection/CollectionEnvelope.java]: @llorllale I'm sorry, but I disagree of this solution. I think the developer/architect/anyone must be free to choose any name he/she want. @yegor256 WDYT about it?

Comment (llorllale) [reply src/main/java/org/cactoos/collection/CollectionEnvelope.java]: @fabriciofx @yegor256 my take: if this is allowed then we should get rid of the `PMD.AbstractNaming` check.

**Comment by 0crat** [comment]
[@llorllale](http://www.0crat.com/u/llorllale) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by llorllale** [comment]
@0crat waiting

**Comment by llorllale** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/cactoos/issues/528#issuecomment-357504600))

@llorllale The impediment was registered successfully

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/cactoos/issues/528#issuecomment-357504603))

@llorllale Job [#528](https://github.com/yegor256/cactoos/issues/528) is already on hold

**Comment by llorllale** [comment]
@yegor256 @fabriciofx I think this job has been on hold for too long. Regarding this solution, I definitely agree on the names @fabriciofx has chosen. My only objection is that we should really remove the `PMD.AbstractNaming` check from the project. Would you agree if I accept these changes but submit a new ticket requesting removal of the check?

**Comment by fabriciofx** [comment]
@llorllale @yegor256 The problem is if `PMD.AbstractNaming` will be removed, Qulice will catch it and doesn't build the app.

**Comment by llorllale** [comment]
@fabriciofx when I said "remove", I meant to *stop checking for this*. Perhaps [excluding rules from the ruleset](http://pmd.sourceforge.net/pmd-4.3.0/howtomakearuleset.html).

**Comment by fabriciofx** [comment]
@llorllale Yeah, I agree, but let's see what @yegor256 thinks.

**Comment by llorllale** [comment]
@fabriciofx I've opened #557 so we can continue discussion on those checks there.

@yegor256 I approve this PR

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Order was successfully finished: +30 points just awarded to @llorllale[/z](http://www.0crat.com/u/llorllale), total is [+75](http://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](http://www.0crat.com/u/yegor256), total is [+7465](http://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#528](https://github.com/yegor256/cactoos/issues/528) is now out of scope

**Pull request merged**